add ISACC user ID to email notification

achen2401 opened the pull request:
https://www.pivotaltracker.com/story/show/185629665
Add existing "ISACC user ID" to notification emails

example ISACC user ID in FHIR Patient resource:
```
"identifier": [
 ...
    {
      "system": "http://isacc.app/user-id", 
      "value": "Test-123"
    }
  ]
```

Commit:
add ISACC user ID to email notification

Commit:
add space

achen2401 commented:
Note, this needs to be tested.  I wasn't able to properly test this as I don't have messaging service set up in my instance.

mcjustin reviewed (approved):
Looks good to me, but I think we need @pbugni 's eyes on this.

pbugni commented on isacc_messaging/api/email_notifications.py:
```diff
@@ -2,22 +2,35 @@
+            extra={'recipient_emails': ' '.join(recipients)},
+            level='error'
+        )
+        return "Need patient"
```
is this happening?  if we are landing in this function w/o a patient, i think the better place to handle is from the caller when the patient is looked up (I don't see a guard there now, and it looks like `get_patient()` could indeed return None.

pbugni committed:
The `get_patient()` accessor will now raise an `IsaccNotFoundError`
exception should a requested patient_id not return a Patient from HAPI.

(none of the existing calls to this method guarded against a failed
lookup, makes more sense to treat such a situation as exceptional)

pbugni committed:
Removed redundant valid patient check.
Include improved message detailing patient ID in email html
